qiskit/transpiler/passes/scheduling/padding/dynamical_decoupling.py

```diff
@@ -227,7 +227,9 @@ def _pre_runhook(self, dag: DAGCircuit):
             for gate in self._dd_sequence:
                 try:
                     # Check calibration.
-                    gate_length = dag.calibrations[gate.name][(physical_index, gate.params)]
+                    gate_length = dag.calibrations[gate.name][
+                        ((physical_index,), tuple(gate.params))
+                    ].duration
```

Comment:
In `QuantumCircuit.has_calibration_for`, the `params` tuple is built by coercing bound `ParameterExpression` objects to `float` as well: https://github.com/Qiskit/qiskit/blob/e5997133d6d8c89405c17e89117216e986a87462/qiskit/circuit/quantumcircuit.py#L451-L456

Does that need to be done here too?

Reply:
I think you are right. I think it would probably be best to have a `QuantumCircuit.get_calibration(gate, qubits, params)` method to abstract this operation, but I won't add that here.

Reply:
Yeah, agreed on both parts.